test/unit/metrics/test_metricframe_streaming.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+import pytest
+import sklearn.metrics as skm
+
+import fairlearn.metrics as metrics
+from test.unit.input_convertors import conversions_for_1d
+from .data_for_test import y_t, y_p, g_4
+from .utils import batchify
+
+
+@pytest.mark.parametrize("transform_y_p", conversions_for_1d)
+@pytest.mark.parametrize("transform_y_t", conversions_for_1d)
+@pytest.mark.parametrize("num_batches", [1, 3, 5])
+def test_sample_params(transform_y_t, transform_y_p, num_batches):
+    sample_weights = np.random.rand(len(y_t))
+    batch_generator = batchify(num_batches, y_t, y_p, g_4, sample_weights)
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 sample_params=None, streaming=True)
+
+    with pytest.raises(ValueError) as excinfo:
```

Comment:
I think this should be a separate test.

Reply:
And on a related point, what happens if you call `.overall` before calling `add_data()` when streaming?

Reply:
It will raise `ValueError('No data to process.')` Exception message to be reworded.